clients.go

```diff
@@ -171,8 +171,63 @@ func NewClient(c net.Conn, o *ops) *Client {
 	return cl
 }
 
+// NewClient returns a new instance of Client.
+func (s *Server) NewClient(c net.Conn) *Client {
+	cl := &Client{
+		Net: ClientConnection{
+			conn:   c,
+			bconn:  bufio.NewReadWriter(bufio.NewReader(c), bufio.NewWriter(c)),
+			Remote: c.RemoteAddr().String(),
+		},
+		State: ClientState{
+			Inflight:      NewInflights(),
+			Subscriptions: NewSubscriptions(),
+			TopicAliases:  NewTopicAliases(s.Options.Capabilities.TopicAliasMaximum),
+			keepalive:     defaultKeepalive,
+		},
+		Properties: ClientProperties{
+			ProtocolVersion: defaultClientProtocolVersion, // default protocol version
+		},
+		ops: &ops{
+			capabilities: s.Options.Capabilities,
+			info:         s.Info,
+			hooks:        s.hooks,
+			log:          s.Log,
+		},
+	}
+
+	cl.refreshDeadline(cl.State.keepalive)
+
+	return cl
+}
+
 // NewInlineClient returns a client used when publishing from the embedding system.
-func NewInlineClient(id, remote string) *Client {
+func (s *Server) NewInlineClient(id, remote string) *Client {
```

Comment:
I like the idea of add new client as a server method, however we shouldn't mix methods and files (eg. server method in client struct file) 👍🏻

Reply:
I like your code and erudition. I will learn from you more in the future. If there is anything I can do for you, I will always be here.